Tasks/PowerShellOnTargetMachines2/PowerShellOnTargetMachines.ps1

```diff
@@ -0,0 +1,43 @@
+Trace-VstsEnteringInvocation $MyInvocation
+$global:ErrorActionPreference = 'Continue'
+
+# Undocumented VstsTaskSdk variable so Verbose/Debug isn't converted to ##vso[task.debug].
+# Otherwise any content the ad-hoc script writes to the verbose pipeline gets dropped by
+# the agent when System.Debug is not set.
+$global:__vstsNoOverrideVerbose = $true
+
+try {
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    . "$PSScriptRoot\Utility.ps1"
+    . "$PSScriptRoot\SessionHelper.ps1"
+    . "$PSScriptRoot\JobHelper.ps1"
+    . "$PSScriptRoot\RunPowerShellScriptJob.ps1"
+
+    # Get all inputs for the task
+    $input_Machines = Get-VstsInput -Name "Machines" -Require -ErrorAction "Stop"
+    $input_UserName = Get-VstsInput -Name "UserName" -Require -ErrorAction "Stop"
```

Comment:
Let's check with @RoopeshNair if we have to make the remote credentials mandatory. We could make this optional and connect to remote using agent account if no specific input is given.

Reply:
Yes you are right. Since the current task also treats these as optional inputs, I see no reason why we should make them mandatory. I will fix this.

Reply:
but for credssp auth, credentials are mandatory. so i will need to throw in case auth is credssp and credentials are not supplied.